nucleus/__init__.py

```diff
@@ -178,7 +178,7 @@ def __init__(
             import tqdm.notebook as tqdm_notebook
 
             self.tqdm_bar = tqdm_notebook.tqdm
-        self._connection = Connection(self.api_key, self.endpoint)
+        self.connection = Connection(self.api_key, self.endpoint)
```

Comment:
Tricky design decision here. `Track` needs access to at least a `Connection` object to be able to call the API to upload track metadata, preferably `NucleusClient` access for consistency with other classes like `Model`. However, a `Validate` instance is created from `NucleusClient`, and so attempting to import `Track` from `ScenarioTest`, from where it is created in `get_items`, will cause a circular import. The lessermost evil was to update `Track.from_json` to take a `Connection` instead of a `NucleusClient`.

The tradeoff here was exposing `connection` as a public property of `NucleusClient` so to access this property when creating `Track`s in cases when only a `NucleusClient` is available. This should be safe, since a `Connection` consists of two public properties––`api_key` and `endpoint`.

Lmk if there are any issues with this approach.

Reply:
Yeah this is fine.

I think passing around a `connection` is a newer approach than passing the client. I actually don't see any reason for a class having to have full access to the client. We should probably refactor everything to pass around the connection object itself.

I don't see any harm in doing things this way so ... let's go ahead 🙂